src/algorithms/classification/extra_trees.py

```diff
@@ -402,3 +404,18 @@ def _get_color(self, label: str) -> str:
         index = np.argwhere(self.labels == label)[0]
         color = [min(1, 1.2 * c) for c in colormap(normalize(index))[0]]
         return matplotlib.colors.to_hex(color)
+
+    def _update_metric(self):
+        self.metrics_info = {}
+        if not len(self.test_data):
+            return
+        true = false = 0
+        for _, row in self.test_data.iterrows():
+            target = row[self.tree_parameters["label_name"]]
+            prediction = self.predict(row.drop(self.tree_parameters["label_name"]))
+            predicted = list(prediction.index)[0]
+            if target == predicted:
+                true += 1
+            else:
+                false += 1
+        self.metrics_info["accuracy"] = round(true / (true + false), 3)
```

Comment:
One variable should be enough just for counting whether the test case passed. You can reuse len(self.test_data) in denominator.